src/modules/navigator/mission_base.cpp

```diff
@@ -367,6 +367,42 @@ MissionBase::on_active()
 	updateAltToAvoidTerrainCollisionAndRepublishTriplet(_mission_item);
 }
 
+bool
+MissionBase::isLanding()
+{
+	if (hasMissionLandStart()) {
```

Comment:
Only thing left is that now this function is called regularly. Currently it will always call the getNextPositionItems, which is searching through the mission items. It will be blocking if it is called at the start of the mission and the mission contains many items because the land mission items will not be in the cache yet. I already proposed a solution here: https://github.com/PX4/PX4-Autopilot/pull/23664/files#diff-e478a3707c309ee3511e233bef53b84bb5b0de134989aad2c7e332a62aa32a34R372
Since we already know the land_start mission idx, as long as the current index is below, we do not need to check it.